client/connector.go

```diff
@@ -287,18 +285,33 @@ func (u *UpsertBatchBuilderImpl) prepareEnumCases(tableName, columnName string,
 			continue
 		}
 
+		abandonRow := false
 		if enumCase, ok := value.(string); ok {
-			if caseInsensitive {
-				enumCase = strings.ToLower(enumCase)
+			if len(enumCase) > defaultStringEnumLength {
+				abandonRow = true
+				u.logger.With(
+					"name", "prepareEnumCases",
+					"error", "Enum string value is too long",
+					"table", tableName,
+					"columnID", columnID,
+					"value", value).Debug("Enum string value is too long")
```

Comment:
we also might want some metrics for this case for monitoring

Reply:
added